include/swift/AST/Identifier.h

```diff
@@ -169,6 +169,15 @@ class Identifier {
     return str().startswith("$") && !(getLength() == 1);
   }
   
+  /// Return true if \c argLabel matches parameter name
+  /// and return false if the tests for valid \c argLabel fail.
+  ///
+  /// The three tests to check if \c argLabel is valid are:
+  /// 1. allows \c argLabel if it matches parameter name,
+  /// 2. allows \c $_  for omitted projected value label,
+  /// 3. allows \c $argLabel if it matches parameter name without its \c $ prefix.
+  bool matchesArgumentLabel(Identifier const &argLabel) const;
```

Comment:
This seems too specific to have on an `Identifier`. Should it this method be a closure in `matchCallArgumentsImpl` instead or a static in CSSimplify.cpp?

Reply:
Heh, I think I suggested that she put this on `Identifier`. It looks like both calls are in `matchCallArgumentsImpl`, though, so a static function above this in CSSimplify.cpp makes sense to me. I think it would also make sense to put on `AnyFunctionType::Param`, but what we have in that closure in `matchCallArugmentsImpl` is just an identifier for the parameter label (which could be changed of course, but seems unnecessary to do for a small stylistic thing).

Reply:
I'm happy to turn this into a static function (or on `AnyFunctionType::Param`) if that's preferable here !

Reply:
Yeah, I think that might be the best spot for it semantically.

Reply:
I still leave in the world where `AnyFunctionType::Param` doesn't exist sometimes... :)